lib/athens/connection.rb

```diff
@@ -8,11 +8,17 @@ class Connection
     def initialize(database: nil, aws_client_override: {})
       @database_name = database
 
-      client_config = {
-        access_key_id: Athens.configuration.aws_access_key,
-        secret_access_key: Athens.configuration.aws_secret_key,
-        region: Athens.configuration.aws_region
-      }.merge(aws_client_override).compact
+      if (Athens.configuration.aws_profile)
```

Comment:
There's no need for the if/else switch here, the `compact` will remove nils from the hash, so you can just set it along with the other fields in the original code